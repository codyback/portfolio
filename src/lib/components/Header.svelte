<script>
  import Hamburger from '@/lib/components/Navigation/Hamburger.svelte';

  export let sidebar = false;
</script>

<header class="flex justify-between border-b">
  <div class="py-2">
    <a href="/" class="font-bold tracking-wider leading-10">
      CODY BACKUS
    </a>
  </div>
  <nav class="py-2 tracking-wider hidden sm:inline">
    <a href="/about" class="px-2 py-3 rounded-md hover:bg-slate-200 leading-10">ABOUT</a>
    <a href="/projects" class="px-2 py-3 rounded-md hover:bg-slate-200 leading-10">PROJECTS</a>
    <a href="/blog" class="px-2 py-3 rounded-md hover:bg-slate-200 leading-10">BLOG</a>
    <a href="/resources" class="px-2 py-3 rounded-md hover:bg-slate-200 leading-10">RESOURCES</a>
  </nav>
  <div class="py-2 text-right w-fit">
    <img class="h-10 w-10 rounded-full bg-slate-500 hidden sm:inline-block" src="" alt="" />
    <Hamburger bind:open={sidebar} />
  </div>

</header>
