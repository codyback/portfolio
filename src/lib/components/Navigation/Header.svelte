<script>
	import Hamburger from '@/lib/components/Navigation/Hamburger.svelte';
	import { page } from '$app/stores';

	export let sidebar = false;
</script>

<header class="flex justify-between border-b z-20 relative">
	<div class="py-2">
		<a
			href="/"
			class="font-bold text-xl leading-10 tracking-tight"
		>
			Cody Backus
		</a>
	</div>

	<div class="py-2 text-right w-fit">
		<nav class="py-2 tracking-wider hidden sm:inline text-neutral-800">
			<a
				href="/about"
				class="px-2 py-3 rounded-md hover:bg-neutral-200 leading-10"
				class:underline={$page.route.id?.startsWith('/about')}
			>
				About
			</a>
			<a
				href="/projects"
				class="px-2 py-3 rounded-md hover:bg-neutral-200 leading-10"
				class:underline={$page.route.id?.startsWith('/projects')}
			>
				Projects
			</a>
			<a
				href="/blog"
				class="px-2 py-3 rounded-md hover:bg-neutral-200 leading-10"
				class:underline={$page.route.id?.startsWith('/blog')}
			>
				Blog
			</a>
			<a
				href="/resources"
				class="px-2 py-3 rounded-md hover:bg-neutral-200 leading-10"
				class:underline={$page.route.id?.startsWith('/resources')}
			>
				Resources
			</a>
		</nav>
		<img
			class="h-10 w-10 rounded-full bg-slate-500 hidden sm:inline-block"
			src=""
			alt=""
		/>
		<Hamburger bind:open={sidebar} />
	</div>
</header>
