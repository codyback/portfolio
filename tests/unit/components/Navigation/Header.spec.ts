import Header from '@/lib/components/Navigation/Header.svelte';
import { render } from '@testing-library/svelte';
import { vi } from 'vitest';

describe('Header', () => {
  beforeAll(() => {
    vi.resetAllMocks();
  });

  it('renders correctly with open false', () => {
    const { container } = render(Header, { sidebar: false });

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with open true', () => {
    const { container } = render(Header, { sidebar: true });

    expect(container).toMatchSnapshot();
  });
});
