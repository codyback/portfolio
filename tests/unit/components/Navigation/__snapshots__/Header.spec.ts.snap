// Vitest Snapshot v1

exports[`Header > renders correctly with open false 1`] = `
<body>
  <div>
    <header
      class="flex justify-between items-center border-b z-20 relative"
    >
      <div
        class="py-2"
      >
        <a
          aria-label="Homepage"
          class="font-bold text-xl leading-10 tracking-tight block hover:scale-110 transition-all duration-300"
          href="/"
          title="Homepage"
        >
          Cody Backus
        </a>
      </div>
       
      <div
        class="py-2 text-right w-fit"
      >
        <nav
          class="py-2 px-4 tracking-wider hidden sm:inline text-neutral-800 dark:text-neutral-300"
        >
          <a
            aria-label="About Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10 underline"
            href="/about"
            title="About Page"
          >
            About
          </a>
          <!--&lt;NavLink&gt;-->
          <a
            aria-label="Projects Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
            href="/projects"
            title="Projects Page"
          >
            Projects
          </a>
          <!--&lt;NavLink&gt;-->
          <a
            aria-label="Blog Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
            href="/blog"
            title="Blog Page"
          >
            Blog
          </a>
          <!--&lt;NavLink&gt;-->
          <a
            aria-label="Resources Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
            href="/resources"
            title="Resources Page"
          >
            Resources
          </a>
          <!--&lt;NavLink&gt;-->
        </nav>
         
        <div
          class="flex sm:inline-block justify-center align-middle items-center"
        >
          <button
            aria-label="Toggle color mode"
            class="sm:px-2 py-1 pt-2 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 box-border px-10"
            title="Toggle color mode"
          >
            <span
              class="dark:hidden inline-block"
            >
              <svg
                fill="#27272a"
                height="24"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM352 256c0 53-43 96-96 96s-96-43-96-96s43-96 96-96s96 43 96 96zm32 0c0-70.7-57.3-128-128-128s-128 57.3-128 128s57.3 128 128 128s128-57.3 128-128z"
                />
              </svg>
            </span>
             
            <span
              class="dark:inline-block hidden"
            >
              <svg
                fill="#d4d4d4"
                height="24"
                viewBox="0 0 384 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.5 32C100 32 0 132.3 0 256S100 480 223.5 480c60.6 0 115.5-24.2 155.8-63.4c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-9.8 1.7-19.8 2.6-30.1 2.6c-96.9 0-175.5-78.8-175.5-176c0-65.8 36-123.1 89.3-153.3c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-6.3-.5-12.6-.8-19-.8z"
                />
              </svg>
            </span>
          </button>
          <!--&lt;ThemeToggle&gt;-->
           
          <button
            class="sm:hidden flex items-center justify-items-end m-auto h-10 w-fit hover:text-gray-700 dark:hover:text-neutral-300 cursor-pointer border-none focus:outline-none s-zPZ177rNd-sa"
          >
            <svg
              class="s-zPZ177rNd-sa"
              height="24"
              width="28"
            >
              <line
                class="s-zPZ177rNd-sa"
                id="top"
                x1="0"
                x2="28"
                y1="3"
                y2="3"
              />
              <line
                class="s-zPZ177rNd-sa"
                id="middle"
                x1="0"
                x2="28"
                y1="11"
                y2="11"
              />
              <line
                class="s-zPZ177rNd-sa"
                id="bottom"
                x1="0"
                x2="28"
                y1="19"
                y2="19"
              />
            </svg>
          </button>
          <!--&lt;Hamburger&gt;-->
        </div>
      </div>
    </header>
    <!--&lt;Header&gt;-->
  </div>
</body>
`;

exports[`Header > renders correctly with open true 1`] = `
<body>
  <div>
    <header
      class="flex justify-between items-center border-b z-20 relative"
    >
      <div
        class="py-2"
      >
        <a
          aria-label="Homepage"
          class="font-bold text-xl leading-10 tracking-tight block hover:scale-110 transition-all duration-300"
          href="/"
          title="Homepage"
        >
          Cody Backus
        </a>
      </div>
       
      <div
        class="py-2 text-right w-fit"
      >
        <nav
          class="py-2 px-4 tracking-wider hidden sm:inline text-neutral-800 dark:text-neutral-300"
        >
          <a
            aria-label="About Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10 underline"
            href="/about"
            title="About Page"
          >
            About
          </a>
          <!--&lt;NavLink&gt;-->
          <a
            aria-label="Projects Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
            href="/projects"
            title="Projects Page"
          >
            Projects
          </a>
          <!--&lt;NavLink&gt;-->
          <a
            aria-label="Blog Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
            href="/blog"
            title="Blog Page"
          >
            Blog
          </a>
          <!--&lt;NavLink&gt;-->
          <a
            aria-label="Resources Page"
            class="px-3 py-3 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 leading-10"
            href="/resources"
            title="Resources Page"
          >
            Resources
          </a>
          <!--&lt;NavLink&gt;-->
        </nav>
         
        <div
          class="flex sm:inline-block justify-center align-middle items-center"
        >
          <button
            aria-label="Toggle color mode"
            class="sm:px-2 py-1 pt-2 rounded-md hover:bg-neutral-200 dark:hover:bg-dark-700 box-border px-10"
            title="Toggle color mode"
          >
            <span
              class="dark:hidden inline-block"
            >
              <svg
                fill="#27272a"
                height="24"
                viewBox="0 0 512 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M361.5 1.2c5 2.1 8.6 6.6 9.6 11.9L391 121l107.9 19.8c5.3 1 9.8 4.6 11.9 9.6s1.5 10.7-1.6 15.2L446.9 256l62.3 90.3c3.1 4.5 3.7 10.2 1.6 15.2s-6.6 8.6-11.9 9.6L391 391 371.1 498.9c-1 5.3-4.6 9.8-9.6 11.9s-10.7 1.5-15.2-1.6L256 446.9l-90.3 62.3c-4.5 3.1-10.2 3.7-15.2 1.6s-8.6-6.6-9.6-11.9L121 391 13.1 371.1c-5.3-1-9.8-4.6-11.9-9.6s-1.5-10.7 1.6-15.2L65.1 256 2.8 165.7c-3.1-4.5-3.7-10.2-1.6-15.2s6.6-8.6 11.9-9.6L121 121 140.9 13.1c1-5.3 4.6-9.8 9.6-11.9s10.7-1.5 15.2 1.6L256 65.1 346.3 2.8c4.5-3.1 10.2-3.7 15.2-1.6zM352 256c0 53-43 96-96 96s-96-43-96-96s43-96 96-96s96 43 96 96zm32 0c0-70.7-57.3-128-128-128s-128 57.3-128 128s57.3 128 128 128s128-57.3 128-128z"
                />
              </svg>
            </span>
             
            <span
              class="dark:inline-block hidden"
            >
              <svg
                fill="#d4d4d4"
                height="24"
                viewBox="0 0 384 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.5 32C100 32 0 132.3 0 256S100 480 223.5 480c60.6 0 115.5-24.2 155.8-63.4c5-4.9 6.3-12.5 3.1-18.7s-10.1-9.7-17-8.5c-9.8 1.7-19.8 2.6-30.1 2.6c-96.9 0-175.5-78.8-175.5-176c0-65.8 36-123.1 89.3-153.3c6.1-3.5 9.2-10.5 7.7-17.3s-7.3-11.9-14.3-12.5c-6.3-.5-12.6-.8-19-.8z"
                />
              </svg>
            </span>
          </button>
          <!--&lt;ThemeToggle&gt;-->
           
          <button
            class="sm:hidden flex items-center justify-items-end m-auto h-10 w-fit hover:text-gray-700 dark:hover:text-neutral-300 cursor-pointer border-none focus:outline-none s-zPZ177rNd-sa open"
          >
            <svg
              class="s-zPZ177rNd-sa"
              height="24"
              width="28"
            >
              <line
                class="s-zPZ177rNd-sa"
                id="top"
                x1="0"
                x2="28"
                y1="3"
                y2="3"
              />
              <line
                class="s-zPZ177rNd-sa"
                id="middle"
                x1="0"
                x2="28"
                y1="11"
                y2="11"
              />
              <line
                class="s-zPZ177rNd-sa"
                id="bottom"
                x1="0"
                x2="28"
                y1="19"
                y2="19"
              />
            </svg>
          </button>
          <!--&lt;Hamburger&gt;-->
        </div>
      </div>
    </header>
    <!--&lt;Header&gt;-->
  </div>
</body>
`;
